pkg/webhook/virtualmachine/virtualmachine.go

```diff
@@ -92,20 +94,29 @@ func (a *virtualMachineAnnotator) Handle(ctx context.Context, req admission.Requ
 
 // mutateCreateVirtualMachinesFn calls the create allocation function
 func (a *virtualMachineAnnotator) mutateCreateVirtualMachinesFn(ctx context.Context, virtualMachine *kubevirt.VirtualMachine) error {
-	log.Info("got a create mutate virtual machine event",
-		"virtualMachineName", virtualMachine.Name,
-		"virtualMachineNamespace", virtualMachine.Namespace)
+	logger := log.WithName("mutateCreateVirtualMachinesFn").WithValues("virtualMachineName", virtualMachine.Name, "virtualMachineNamespace", virtualMachine.Namespace)
+	logger.V(1).Info("got a create mutate virtual machine event")
 
 	existingVirtualMachine := &kubevirt.VirtualMachine{}
 	err := a.client.Get(context.TODO(), client.ObjectKey{Namespace: virtualMachine.Namespace, Name: virtualMachine.Name}, existingVirtualMachine)
 	if err != nil {
 		// If the VM does not exist yet, allocate a new MAC address
-		if errors.IsNotFound(err) {
-			return a.poolManager.AllocateVirtualMachineMac(virtualMachine)
+		if apierrors.IsNotFound(err) {
+			err = a.poolManager.AllocateVirtualMachineMac(virtualMachine)
+			if err != nil {
+				return errors.Wrap(err, "Failed to allocate mac to the vm object")
+			}
+
+			if virtualMachine.ObjectMeta.DeletionTimestamp.IsZero() {
```

Comment:
Since this is at webhook we don't need this anymore.

Reply:
I added this after reading this [discussion](https://github.com/knative/eventing/issues/643#issuecomment-443017409) on adding finalizers on webhooks.
The scenario is that a possibly the vm is already being deleted by **another** webhook who had rejected it. so basically if it is already being deleted then no use to add our finalizer.. 

but when I think of it, maybe also the allocation should be rejected..